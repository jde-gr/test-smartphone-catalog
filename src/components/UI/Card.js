import React from 'react';
import { CardContainer } from './styles/Card.styles';

const Card = (props) => {
  return <CardContainer>{props.children}</CardContainer>;
};

export default Card;
