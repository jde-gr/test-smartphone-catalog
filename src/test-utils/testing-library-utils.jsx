import { render } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../store';

const queryClient = new QueryClient();

const AllTheProviders = ({ children }) => {
  return (
    <BrowserRouter>
      <Provider store={store}>
        <QueryClientProvider client={queryClient}>
          {children}
        </QueryClientProvider>
      </Provider>
    </BrowserRouter>
  );
};
const renderWithContext = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { renderWithContext as render };
