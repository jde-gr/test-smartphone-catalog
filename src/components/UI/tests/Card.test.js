import { render } from '../../../test-utils/testing-library-utils';
import Card from '../Card';

test('renders correctly', () => {
  const { container } = render(<Card />);
  expect(container).toMatchSnapshot();
});
