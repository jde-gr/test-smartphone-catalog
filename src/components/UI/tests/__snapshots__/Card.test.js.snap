// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    style="width: inherit;"
  >
    <div
      class="css-1xb5uqo"
    />
  </div>
</div>
`;
